internal/tfimport/importer/resources_test.go

```diff
@@ -164,3 +167,55 @@ func TestSimpleImporterErr(t *testing.T) {
 		}
 	}
 }
+
+func TestLoadFields(t *testing.T) {
+	tests := []struct {
+		fields []string
+		want   map[string]string
+	}{
+		// Empty.
+		{[]string{}, map[string]string{}},
+
+		// Get some fields from both locations.
+		{
+			[]string{"", "name"},
+			map[string]string{
+				"":     "emptyValFirst",
+				"name": "myresourcename",
+			},
+		},
+	}
+	for _, tc := range tests {
+		rc := resourceChange.Change.After
+		got, err := loadFields(tc.fields, false, configs[0], rc)
+		if err != nil {
+			t.Fatalf("loadFields(%v, %v, %v, %v) failed: %v", tc.fields, false, configs[0], rc, err)
+		}
+		if !cmp.Equal(got, tc.want, cmpopts.EquateEmpty()) {
+			t.Errorf("loadFields(%v, %v, %v, %v) = %v; got %v", tc.fields, false, configs[0], rc, got, tc.want)
+		}
+	}
+}
+
+func TestLoadFieldsErr(t *testing.T) {
+	tests := []struct {
+		fields []string
+	}{
+		// Multiple missing fields
+		{[]string{"missing1", "missing2"}},
+	}
+	for _, tc := range tests {
+		rc := resourceChange.Change.After
+		got, err := loadFields(tc.fields, false, configs[0], rc)
+		if err == nil {
+			t.Errorf("loadFields(%v, %v, %v, %v) succeeded for malformed input, want error; got = %v", tc.fields, false, configs[0], rc, got)
+		}
+		var ie *InsufficientInfoErr
```

Comment:
optional: maybe make a wantErr and just do cmp.Diff on the error itself?

Reply:
Done.